explainer.md

```diff
@@ -113,41 +121,60 @@ The CSS images and computed properties/styles cached from Page-A are represented
+The developer can associate particular elements from Page-A to elements from Page-B. This would usually be done if they're equivalent. In this case, the headers, share buttons, and roots are equivalent. When this happens, the state from the Page-B element is added to the same states wrapper:
 
 ```
 transition element
-└─ image wrapper
-   ├─ image (Page-A)
-   └─ image (Page-B)
+└─ states
+   ├─ state (Page-A)
+   │  ├─ image
+   │  └─ …child transition elements…
+   └─ state (Page-B)
+      ├─ image
+      └─ …child transition elements…
 ```
 
-This allows for the container to be moved as one, while cross-fading the Page-A and Page-B content. The developer will also have access to the state of shared elements (from Page-A and Page-B) replicated on the container. This state depends on the capture mode (single image vs computed styles + content image).
+This allows for the container to be moved as one, while cross-fading the Page-A and Page-B states. The developer will also have access to the style state of offered elements (from Page-A and Page-B) replicated on the container. This style state depends on the capture mode (single image vs computed styles + content image).
 
 Transition elements don't need to be associated with another transition elements, which allows for transitions involving elements that are only in Page-A or only in Page-B.
 
 The root elements of each page are automatically associated.
 
 Note that the order in which the transition elements are painted can be configured by UA and/or developer stylesheets using z-index.
 
-- [Open question](https://github.com/WICG/shared-element-transitions/issues/23): How should the default UA animation order these elements? And also handle a change in associated elements between the 2 pages. 
+- [Open question](https://github.com/WICG/shared-element-transitions/issues/23): How should the default UA animation order these elements? And also handle a change in associated elements between the 2 pages.
+
+### How are transition elements painted?
+
+During the transition a new stacking context (called uber-root) is created with the following hierarchy :
+
+```
+uber-root stacking context
+├─ root stacking context
+└─ transition stacking context
+```
+
+This allows using the output of the root stacking context to provide a live root image for Page-B. The transition elements are painted into the transition stacking context. An alternate approach to this was to paint the transition elements in the [top layer](https://fullscreen.spec.whatwg.org/#top-layer). But that made it difficult to support transitions when there is other content in the top layer (fullscreen elements, dialog) and to ensure effects on the root element which are applied to the root stacking context (background-color, filter) are captured in the root image. See [issue](https://github.com/WICG/shared-element-transitions/issues/74) for detailed discussion.
```

Comment:
```suggestion
This allows using the output of the root stacking context to provide a live root image for Page-B. The transition stacking context maps to the transition-root element. An alternate approach to this was to paint the transition-root in the [top layer](https://fullscreen.spec.whatwg.org/#top-layer). But that made it difficult to support transitions when there is other content in the top layer (fullscreen elements, dialog) and to ensure effects on the root element which are applied to the root stacking context (background-color, filter) are captured in the root image. See [issue](https://github.com/WICG/shared-element-transitions/issues/74) for detailed discussion.
```

Can clarify what the transition stacking context is if we specify the transition-root element in the DOM structure.

Reply:
fwiw, I didn't edit this section. I'm not sure I fully understand it.